futures/examples/functional.rs

```diff
@@ -0,0 +1,57 @@
+#![feature(futures_api, async_await, await_macro)]
+#[macro_use] extern crate futures;
+
+use futures::{
+    channel::mpsc,
+    executor, //standard executors to provide a context for futures and streams
+    futures::ready,
+    StreamExt
+};
+
+fn main() {
+    let (tx, rx) = mpsc::unbounded::<i32>();
+
+    //
+    // Create a future by building an async block that will be implemented as a future,
```

Comment:
To be precise: An async block is an expression that gives back a type that implements `Future`

Reply:
How I would phrase it: Create a future via an async block

Reply:
It's probably better to mention `Future::poll` only later